Neos.Utility.Arrays/Tests/Unit/ArraysTest.php

```diff
@@ -108,6 +108,103 @@ public function getValueByPathReturnsNullIfThePathHasMoreSegmentsThanTheGivenArr
         self::assertNull(Arrays::getValueByPath($array, ['Foo', 'Bar', 'Baz', 'Bux']));
     }
 
+    /**
+     * @test
+     */
+    public function getAccessorByPathReturnsTheValueOfANestedArrayByFollowingTheGivenSimplePath()
+    {
+        $array = ['Foo' => 'the value'];
+        self::assertSame('the value', Arrays::getAccessorByPath($array, ['Foo'])->string());
+    }
+
+    /**
+     * @test
+     */
+    public function getAccessorByPathReturnsTheValueOfANestedArrayByFollowingTheGivenPath()
+    {
+        $array = ['Foo' => ['Bar' => ['Baz' => [2 => 'the value']]]];
+        self::assertSame('the value', Arrays::getAccessorByPath($array, ['Foo', 'Bar', 'Baz', 2])->string());
+    }
+
+    /**
+     * @test
+     */
+    public function getAccessorByPathReturnsTheValueOfANestedArrayByFollowingTheGivenPathIfPathIsString()
+    {
+        $array = ['Foo' => ['Bar' => ['Baz' => [2 => 'the value']]]];
+        self::assertSame('the value', Arrays::getAccessorByPath($array, 'Foo.Bar.Baz.2')->string());
+    }
+
+    /**
+     * @test
+     */
+    public function getAccessorByPathThrowsTypeErrorIfPathIsNoArrayOrString()
+    {
+        $this->expectException(\TypeError::class);
+        $array = ['Foo' => ['Bar' => ['Baz' => [2 => 'the value']]]];
+        Arrays::getAccessorByPath($array, null);
+    }
+
+    /**
+     * @test
+     */
+    public function getAccessorByPathReturnsNullIfTheSegementsOfThePathDontExist()
+    {
+        $array = ['Foo' => ['Bar' => ['Baz' => [2 => 'the value']]]];
+        self::assertNull(Arrays::getAccessorByPath($array, ['Foo', 'Bar', 'Bax', 2])->intOrNull());
+    }
+
+    /**
+     * @test
+     */
+    public function getAccessorByPathReturnsNullIfThePathHasMoreSegmentsThanTheGivenArray()
+    {
+        $array = ['Foo' => ['Bar' => ['Baz' => 'the value']]];
+        self::assertNull(Arrays::getAccessorByPath($array, ['Foo', 'Bar', 'Baz', 'Bux'])->intOrNull());
+    }
+
+    /**
+     * @test
+     */
+    public function getAccessorByPathTypeErrorContainsPathForNonMatchingTypes()
+    {
+        $array = ['Foo' => ['Bar' => ['Baz' => 'the value']]];
+        try {
+            self::assertNull(Arrays::getAccessorByPath($array, ['Foo', 'Bar', 'Baz'])->int());
+        } catch (\Error $error) {
+            self::assertInstanceOf(\TypeError::class, $error);
+            self::assertStringEndsWith('in path Foo.Bar.Baz', $error->getMessage());
+        }
```

Comment:
The `self::assertNull()` is probably not evaluated.
Also, instead of `try/catch` you could use the built-in methods to assert exceptions:
```suggestion
        $this->expectException(\TypeError::class);
        $this->expectExceptionMessage('in path Foo.Bar.Baz');
        Arrays::getAccessorByPath($array, ['Foo', 'Bar', 'Baz'])->int();
```

Reply:
adjusted as requested